### PR TITLE
ledger-tool verify cli for final hash calc

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
#31150

#### Summary of Changes
Move final hash calc to cli arg.

Fixes #

### CODE REVIEW

File: ledger-tool/src/main.rs [hunk 1]
```diff
@@ -2928,7 +2934,7 @@ fn main() {
                     on_halt_store_hash_raw_data_for_debug: arg_matches
                         .is_present("halt_at_slot_store_hash_raw_data"),
                     // ledger tool verify always runs the accounts hash calc at the end of processing the blockstore
-                    run_final_accounts_hash_calc: true,
```
Comment (steviez) [inline ledger-tool/src/main.rs:2931]: Can you please rip out the comment on the line above as well that would now be incorrect

**Comment by steviez** [comment]
I'm good with disabling the final hash calc by default, but just to make sure we're on the same page, is this meant to be a quick workaround, and we would leave the issue I created open (potentially addressing at some later point) as a reminder that we could hit this assertion if running the final hash calc ?

**Comment by jeffwashington** [comment]
> What does dealing with the race condition look like?

I've never hit this and I use ledger-tool verify a lot. Dealing with it means that I as someone who passes the new cli arg sees it and I expect it is a race condition, so I comment out asserts or make abs not flush or something.

**Comment by jeffwashington** [comment]
> sharp-edges / gotchas

Yes, this is a sharp edge. I constantly found myself in need of this functionality. Even now it is useful to make sure that I'm not missing something in ledger-tool verify runs. If the final hash and cap matches between different builds, then I know NOTHING about accounts is wrong with a simple compare.

**Comment by steviez** [comment]
> I've never hit this and I use ledger-tool verify a lot. Dealing with it means that I as someone who passes the new cli arg sees it and I expect it is a race condition, so I comment out asserts or make abs not flush or something.

Thoughts on adding a comment and/or adjusting help message to say something along the lines of "Final hash calculation could race with accounts background service tasks"? I have no doubt you know exactly how to mitigate this if you encounter it (and me too because I know to message you if I'm really stuck 😉), but thinking about the everyone else case and trying to leave a little something to steer in the right direction.

I'm definitely not disputing the usefulness of the arg for debugging, but maybe I'm being paranoid that someone who isn't very in the know will be running `verify` at all, let alone trying out various optional command.

**Review by steviez — DISMISSED** [review]
I'm onboard for shipping this - you mentioned it is a debug step, so I think disable by default makes sense.

As a later step, we can evaluate if there is low hanging-fruit to address the actual race condition. If not, maybe just drop a comment or warning

**Review by steviez — APPROVED** [review]

**Pull request merged**